### PR TITLE
HParams: Separate data table controls

### PR DESCRIPTION
**Pull request by JamesHollyer**
## Motivation for features / changes
We have realized we need more direct control over which columns have which controls. This change moves those controls to a set of optional booleans in the ColumnHeader interface and utilizes them instead of a general "ControlsEnabled" boolean which we used before.  Functionally the only change here should be that the runs column cannot be removed anymore. However, this more direct control will help a lot as other columns are added. 

It also fixes a bug where header which could not sort were still dispatching the sorting action.

## Technical description of changes
Added these as optional booleans and started using them instead of the ControlsEnabled boolean.  Since these headers are stored in localStorage already for the Scalar Table I also had to go set them as a default in the persistent settings data source.

## Screenshots of UI changes (or N/A)
Note the lack of a remove button:
run_with_remove

## Detailed steps to verify changes work correctly (as executed by you)
I had old Single and Range headers stored in localStorage. They updated properly now and work great. I messed with the runs table too. It all seems to work. Except I found that ordering columns was never actually hooked up. I will fix that in a follow up.

## Alternate designs / implementations considered (or N/A)
Considered just splitting the controlsEnabled prop to multiple props...This seemed cleaner.

### CODE REVIEW

**Review by rileyajones — COMMENTED** [review]
This mostly looks good but there are a handful of test which still need to be updated

File: tensorboard/webapp/widgets/data_table/data_table_component.ts [hunk 1]
```diff
@@ -271,10 +271,17 @@ export class DataTableComponent implements OnDestroy, AfterContentInit {
...
+    );
+  }
+
+  canContextMenuInsert() {
+    return (
+      this.selectableColumns &&
+      this.selectableColumns.length &&
       this.contextMenuHeader !== undefined &&
-      this.contextMenuHeader.type === ColumnHeaderType.HPARAM
+      this.contextMenuHeader.movable
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ts]: Done.

File: tensorboard/webapp/widgets/data_table/data_table_component.ts [hunk 2]
```diff
@@ -271,10 +271,17 @@ export class DataTableComponent implements OnDestroy, AfterContentInit {
   }
 
   canContextMenuRemoveColumn() {
-    // TODO(rileyajones) Update this check to reply upon a new field on the header type.
     return (
+      this.contextMenuHeader !== undefined && this.contextMenuHeader.removable
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ts]: Done.

File: tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts [hunk 3]
```diff
@@ -165,21 +165,6 @@ describe('runs_data_table', () => {
   });
 
   describe('color column', () => {
-    it('disables controls for color header', () => {
```
Comment (JamesHollyer) [inline tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts:168]: These controls are now disabled by default.

File: tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts [hunk 4]
```diff
@@ -241,21 +226,6 @@ describe('runs_data_table', () => {
     });
   });
 
-  it('disables controls for selected header', () => {
```
Comment (JamesHollyer) [inline tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts:244]: These controls are now disabled by default.

**Review by rileyajones — APPROVED** [review]

**Pull request merged**